### PR TITLE
fix(route53): remove `http://` from bucket target

### LINKED ISSUE #3928: route53 A record is setting incorrect alias target for s3 website
### What is the problem?
I created a cdk stack with s3, route53 and A record on an existing zone. when deployed the alias target in route53 is encoded string

`http\072\057\057harmohan.<website>.com.s3-website-ap-southeast-2.amazonaws.com.`

### Reproduction Steps
```
 const bucketProps: BucketProps = {
      bucketName: DOMAIN_NAME,
      websiteIndexDocument: 'index.html',
      publicReadAccess: true,
      removalPolicy: cdn ? RemovalPolicy.RETAIN : RemovalPolicy.DESTROY,
      websiteErrorDocument: cdn ? undefined : 'index.html'
    }

    const bucket = new Bucket(this, 'S3Bucket', bucketProps);

    const zone = HostedZone.fromLookup(this, 'HostedZone', {
      domainName: <domain name>
    });

    new ARecord(this, 'ARecord', {
      recordName: DOMAIN_NAME,
      zone: zone,
      target: {
        aliasTarget: new BucketWebsiteTarget(bucket)
      }
    });
```

### Environment

  - **CDK CLI Version:** 1.6.0
  - **Module Version:** 1.6.0
  - **OS:** 
  - **Language:** TypeScript

### Other information

Screen Shot 2019-09-04 at 2 00 30 pm

### PR DESCRIPTION
**Pull request by nmussy**
* change bucket type to IBucket to allow static import

Fixes #3928

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-route53-targets/lib/bucket-website-target.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import {RegionInfo} from '@aws-cdk/region-info';
  * Use a S3 as an alias record target
  */
 export class BucketWebsiteTarget implements route53.IAliasRecordTarget {
-  constructor(private readonly bucket: s3.Bucket) {
+  constructor(private readonly bucket: s3.IBucket) {
```
Comment (nmussy) [inline packages/@aws-cdk/aws-route53-targets/lib/bucket-website-target.ts]: Would this be considered a breaking change, knowing that `Bucket extends BucketBase implements IBucket`?

Comment (eladb) [reply packages/@aws-cdk/aws-route53-targets/lib/bucket-website-target.ts]: No because IBucket is more general than Bucket.

**Comment by hoegertn** [comment]
Are you sure the regional domain same has the same semantic as the website url with regards to website hosting?

**Comment by nmussy** [comment]
Good catch @hoegertn, you're right, [it doesn't include `s3-website-`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket.html#RegionalDomainName-fn::getatt). I was about to mess it up again, thanks.

I'll look into it truncating the `websiteURL` property a little later then. @eladb Can you block this PR in the meantime?

**Comment by hoegertn** [comment]
I am currently working on a PR for r53 redirects. Within this pr I added a websiteDomainName attribute to the bucket. Maybe this would help once it is merged...

**Comment by nmussy** [comment]
Yeah, [`websiteDomainName`](https://github.com/aws/aws-cdk/blob/8781042a4a3e57dde83c90dd9d90174a0f05e4f4/packages/%40aws-cdk/aws-s3/lib/bucket.ts#L37) from #3946 looks perfect. I'll hold up until it's merged, thanks again.

**Comment by nmussy** [comment]
@eladb Should be good for review 👍

**Comment by hoegertn** [comment]
I am not sure about the second test. SHouldn't the Join contain `http://`? Otherwise, the split will fail...

**Comment by nmussy** [comment]
@hoegertn Would that an issue with #3946 then? Because I'm only using `fromBucketName` there.

**Comment by hoegertn** [comment]
You are right this is a bug in the Bucket, and I just used the existing `websiteUrl` property but it is incorrect for imported buckets. see #4113

**Pull request merged**